### ISSUE TITLE
`jax.grad` of `jnp.ldexp(x, 1)` returns `1.0` instead of `2.0` at `x=0`

### ISSUE THREAD

**Issue by AjBrown89** [jax-ml/jax#37992]
### Description

The gradient of `jnp.ldexp(x, 1)` with respect to `x` should be `2.0` everywhere, since `ldexp(x, 1) = x * 2^1 = 2x` and `d/dx(2x) = 2`. At `x = 0.0`, JAX returns `1.0`.

```python
import jax, jax.numpy as jnp
import numpy as np
jax.config.update('jax_enable_x64', True)

grad_fn = jax.grad(lambda x: jnp.sum(jnp.ldexp(x, 1)))

print(grad_fn(jnp.array([0.0], dtype=np.float32)))  # [1.]  ← bug, expected [2.]
print(grad_fn(jnp.array([1.0], dtype=np.float32)))  # [2.]  ← correct
print(grad_fn(jnp.array([2.0], dtype=np.float32)))  # [2.]  ← correct
```

The bug triggers only at exactly `x = 0.0`.

**Root cause:** The `ldexp` implementation uses `_where(isinf(x1) | (x1 == 0), x1, computed_result)` to preserve signed zero. At `x = 0`, this selects `x1` directly rather than the computed result. Autodiff therefore flows gradient through `x1` (giving `∂x1/∂x1 = 1`) instead of through the full `ldexp` computation (which correctly gives `2`).

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:    0.10.1.dev20260513+64f6e229b
jaxlib: 0.10.0
numpy:  2.4.4
python: 3.12.3 (main, Mar 23 2026, 19:04:32) [GCC 13.3.0]
device info: cpu-1, 1 local devices
process_count: 1
platform: uname_result(system='Linux', node='eb2-3228-lin08.csc.ncsu.edu', release='6.14.0-37-generic', version='#37~24.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Nov 20 10:25:38 UTC 2', machine='x86_64')
```

**Comment by jakevdp** [comment]
This falls under #38651; please note that it may take a while for this to receive attention from the team.

**Comment by JasonHonKL** [comment]
@jakevdp mind if assign me to try this task ?
